static/app/actionCreators/metrics.tsx

```diff
@@ -0,0 +1,75 @@
+import {Client} from 'sentry/api';
+import {getInterval} from 'sentry/components/charts/utils';
+import {DateString, MetricsApiResponse, Organization} from 'sentry/types';
+import {getPeriod} from 'sentry/utils/getPeriod';
+
+export type DoMetricsRequestOptions = {
+  field: string[];
+  orgSlug: Organization['slug'];
+  cursor?: string;
+  end?: DateString;
+  environment?: Readonly<string[]>;
+  groupBy?: string[];
+  includeAllArgs?: boolean;
+  interval?: string;
+  limit?: number;
+  orderBy?: string;
+  project?: Readonly<number[]>;
+  query?: string;
+  start?: DateString;
+  statsPeriod?: string | null;
+  statsPeriodEnd?: string | null;
+  statsPeriodStart?: string | null;
+};
+
+export const doMetricsRequest = (
+  api: Client,
+  {
+    field,
+    orgSlug,
+    cursor,
+    environment,
+    groupBy,
+    interval,
+    limit,
+    orderBy,
+    project,
+    query,
+    includeAllArgs = false,
+    statsPeriodStart,
+    statsPeriodEnd,
+    ...dateTime
+  }: DoMetricsRequestOptions
+): Promise<MetricsApiResponse> => {
+  const {start, end, statsPeriod} =
+    statsPeriodStart && statsPeriodEnd
+      ? {start: undefined, end: undefined, statsPeriod: undefined}
+      : getPeriod({
+          period: dateTime.statsPeriod,
+          start: dateTime.start,
+          end: dateTime.end,
+        });
```

Comment:
Should we be handling this logic of what values to set vs what to set as undefined here? If the backend can handle which params take priority (not 100% sure if it does that), should we just pass all the params we get here and let the backend decide which one to apply?

Reply:
I replaced `getPeriod` with `normalizeDateTimeParams` that supports empty statsPeriod. Now it should be much cleaner while addressing your comment.